feat(core)!: Set the `secure` flag on issued cookies

netroy opened the pull request:
Currently n8n does not set `secure` flag on issues cookies. This means that if an instance is reachable over http, it could be vulnerable to MITM attacks.

N8N-7246

## Review / Merge checklist
- [x] PR title and summary are descriptive
- [x] Tests included

netroy committed:
feat(core): Set the `secure` flag on issued cookies

tomi reviewed (dismissed):
🚀

Should we mark this as a breaking change? As it does break the app if someone is hosting it over http.

netroy commented:
Updated `BREAKING-CHANGES.md` and the PR title to mark this as a breaking changes, and added a new config variable to let people disable the `Secure` flag.

tomi commented on packages/cli/src/config/index.ts:
```diff
@@ -15,6 +15,7 @@ if (inE2ETests) {
 	process.env.N8N_LOG_LEVEL = 'silent';
 	process.env.N8N_PUBLIC_API_DISABLED = 'true';
 	process.env.SKIP_STATISTICS_EVENTS = 'true';
+	process.env.N8N_SECURE_COOKIE = 'false';
```
Maybe we could set it to true in tests so it corresponds the most common config? Or leave it out so the default value gets tested

netroy replied:
we unfortunately can't to do that right now because `superagent` doesn't handle secure cookies, which breaks every integration test that updates the cookie in response.

tomi reviewed (approved):
🚀

janober commented:
Got released with [`n8n@1.31.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.31.1)
